PowerShellTools/DebugEngine/Remote/IPowerShellToolsPortSupplier.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Debugger.Interop;
+
+namespace PowerShellTools.DebugEngine.Remote
+{
+    /// <summary>
+    /// Interface for all remote debugging port suppliers that this tool implements. Enforces the
+    /// need for said port suppliers to specify whether or not the -UseSSL parameter should be used
+    /// when entering into a remote session with a machine.
+    /// </summary>
+    interface IPowerShellToolsPortSupplier : IDebugPortSupplier2, IDebugPortSupplierDescription2
+    {
+        bool UsesSSL();

```

Comment:
You can't have private properties in an interface, and whether or not a port supplier uses SSL should not be modifiable by someone else. It also allows for flexibility for a future supplier to define whether or not they use SSL. Maybe for them it won't be a true/false situation, it might be based one some conditions, etc. Lastly, I think using a method is more true to the purpose of an interface which is to provide a contract that other objects can use to communicate with your object. All of that make sense?